tests/test_misc.py

```diff
@@ -28,5 +30,43 @@ def test_url_pat_3(self):
         self.assertEqual(m, None)
 
 
+def make_mock_directory(tmpdir, mock_directory):
+    for key, value in mock_directory.items():
+        if value is None:
+            tmpdir.join(key).write("TEST")
+        else:
+            make_mock_directory(tmpdir.mkdir(key), value)
+
+def test_walk_prefix(tmpdir):  # tmpdir is a py.test utility
+    # Each directory is a dict whose keys are names. If the value is
+    # None, then that key represents a file. If it's another dict, that key is
+    # a file
+    mock_directory = {
+        "LICENSE.txt": None,
+        "envs": {"ignore1": None,
+                 "ignore2": None},
+        "python.app": None,
+        "bin": {"activate": None,
+                "conda": None,
+                "deactivate": None,
+                "testfile": None},
+        "testdir1": {"testfile": None,
+                     "testdir2": {"testfile": None}},
+        "testfile1": None,
+    }
+
+    make_mock_directory(tmpdir, mock_directory)
```

Comment:
Fine as far as I'm concerned.  Actually great.  We're only at ~50% coverage right now.  The more tests the better.